Microsoft.Toolkit.Uwp.UI.Controls/Carousel/CarouselItemAutomationPeer.cs

```diff
@@ -0,0 +1,147 @@
+            }
+        }
+
+        private CarouselItem OwnerCarouselItem
+        {
+            get { return this.Owner as CarouselItem; }
+        }
+
+        /// <summary>Adds the current element to the collection of selected items.</summary>
+        public void AddToSelection()
+        {
+            CarouselItem owner = this.OwnerCarouselItem;
+            Carousel parent = owner.ParentCarousel;
+            parent?.SetSelectedItem(owner);
+        }
+
+        /// <summary>Removes the current element from the collection of selected items.</summary>
+        public void RemoveFromSelection()
+        {
+            CarouselItem owner = this.OwnerCarouselItem;
+            Carousel parent = owner.ParentCarousel;
+            if (parent != null)
+            {
+                parent.SelectedItem = null;
+            }
+        }
+
+        /// <summary>Clears any existing selection and then selects the current element.</summary>
+        public void Select()
+        {
+            CarouselItem owner = this.OwnerCarouselItem;
+            Carousel parent = owner.ParentCarousel;
+            parent?.SetSelectedItem(owner);
+        }
+
+        /// <summary>
+        /// Gets the control type for the element that is associated with the UI Automation peer.
+        /// </summary>
+        /// <returns>The control type.</returns>
+        protected override AutomationControlType GetAutomationControlTypeCore()
+        {
+            return AutomationControlType.ListItem;
+        }
+
+        /// <summary>
+        /// Called by GetClassName that gets a human readable name that, in addition to AutomationControlType,
+        /// differentiates the control represented by this AutomationPeer.
+        /// </summary>
+        /// <returns>The string that contains the name.</returns>
+        protected override string GetClassNameCore()
+        {
+            return Owner.GetType().Name;
+        }
+
+        /// <summary>
+        /// Called by GetName.
+        /// </summary>
+        /// <returns>
+        /// Returns the first of these that is not null or empty:
+        /// - Value returned by the base implementation
+        /// - Name of the owning CarouselItem
+        /// - Carousel class name
+        /// </returns>
+        protected override string GetNameCore()
+        {
+            int? index = this.OwnerCarouselItem.ParentCarousel?.IndexFromContainer(this.OwnerCarouselItem);
+
+            string name = base.GetNameCore();
+            if (!string.IsNullOrEmpty(name))
+            {
+                return $"{name}";
+            }
+
+            if (this.OwnerCarouselItem != null && !string.IsNullOrEmpty(this.OwnerCarouselItem.Name))
+            {
+                return this.OwnerCarouselItem.Name;
+            }
+
+            if (string.IsNullOrEmpty(name))
+            {
+                name = this.GetClassName();
+            }
+
+            return $"{name} {index}";
```

Comment:
I think using the AutomationProprties.SizeOfSet property is the better way to do this then hardcoding it in the name.

Reply:
Removed this, I don't think this was actually needed. But I have implemented the size of set and position in set methods.

Reply:
@jamesmcroft I didn't see the new size/position code just 'string.Empty' think it's still useful to return that if we have nothing else.

Reply:
The problem is, if a name is provided, it will return "My Name 1" which isn't actually what you'd want. The size was added in originally because we were returning the class name so it would have been "CarouselItem 1". Based on the feedback, the class name has been removed so not sure this was needed. 

I can always add the class name back in though or if nothing else, return the index.